components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIManagerConfiguration.java

```diff
@@ -181,8 +181,20 @@ else if("Environments".equals(localName)) {
                     environment.setPassword(replaceSystemProperty(value));
                     environment.setApiGatewayEndpoint(replaceSystemProperty(
                             environmentElem.getFirstChildWithName(new QName(
-                                    APIConstants.API_GATEWAY_ENDPOINT)).getText()));
-                    apiGatewayEnvironments.add(environment);
+                                                            APIConstants.API_GATEWAY_ENDPOINT)).getText()));
+                    OMElement description =
+                            environmentElem.getFirstChildWithName(new QName("Description"));
+                    if (description != null) {
+                        environment.setDescription(description.getText());
+                    } else {
+                        environment.setDescription("");
+                    }
+                    if (!apiGatewayEnvironments.containsKey(environment.getName())) {
+                        apiGatewayEnvironments.put(environment.getName(), environment);
+                    } else {
```

Comment:
Put a comment to indicate that this occurs during server startup and hence the reason we're logging and continuing.